allure-kotlin-migration-guide.md

```diff
@@ -0,0 +1,94 @@
+       androidTestImplementation "io.qameta.allure:allure-android-commons:$ALLURE_ANDROID_VERSION"
+  
+       //ADD allure-kotlin dependencies
+       androidTestImplementation "io.qameta.allure:allure-kotlin-commons:$ALLURE_KOTLIN_VERSION"
+       androidTestImplementation "io.qameta.allure:allure-kotlin-junit4:$ALLURE_KOTLIN_VERSION"
+   }
+    ```
+1. Migrate core classes
+    1. Migrate global `step` function from `io.qameta.allure.android.step` to `io.qameta.allure.kotlin.Allure#step` using the following example (the API is a bit different and allows for specifying parameters inside the lambda):
+        ```kotlin
+        //allure-android definition
+        step("Step1", Parameter("param1", "true"), Parameter("param2", "false")) {
+            assertTrue(true)
+        }
+        ...
+        //allure-kotlin definition
+        Allure.step("Step 1") {
+            parameter("param1", "true")
+            parameter("param2", "false")
+            assertTrue(true)
+        }
+        ```
+        Explore `Allure` API as it allows to define labels, epics and other allure components from the code, instead of only via annotations.
+    1. Migrate `AllureRunListener` from `io.qameta.allure.android.AllureRunListener` -> `io.qameta.allure.kotlin.junit4.AllureJunit4`
+    1. Migrate `AllureLifecycle` from `io.qameta.allure.android.AllureLifecycle` -> `io.qameta.allure.kotlin.AllureLifecycle` (which has a lot more capabilities and allows to attach lifecycle listeners)
+    1. Change import for `AllureRunner` from `io.qameta.allure.android.AllureRunner` -> `io.qameta.allure.kotlin.junit4.AllureRunner`
+1. Migrate package `annotations`
+    1. Change all imports from `io.qameta.allure.android.annotations.*` -> `io.qameta.allure.kotlin.*` (explore new annotations package as support for many more annotations has been added)
+    1. Change imports for Allure Junit4 annotations `DisplayName`, `Tag`, `Tags` from `io.qameta.allure.android.annotations.*` -> `io.qameta.allure.kotlin.junit4.*`
+    1. Change import for `SeverityLevel` from `io.qameta.allure.android.SeverityLevel` -> `io.qameta.allure.kotlin.SeverityLevel`
+1. Migrate package `io`
+    1. Change all imports from `io.qameta.allure.android.io.*` -> `io.qameta.allure.kotlin.*`
+    1. In case you have custom `AllureResultsWriter` adjust it as the API has changed a bit
+    1. `AllureResultsReader` & `FileSystemResultsReader` have been removed as they are not used anymore
+1. Migrate package `listener` - change all imports from `io.qameta.allure.android.listener.*` -> `io.qameta.allure.kotlin.listener.*`
+1. Migrate package `serialization` - it was used internally and is no longer needed.
+1. Migrate package `utils` - it was used to parse the annotations to allure results. It shouldn't be accesed by consumer code. If you were using any of those methods take a look at ` io.qameta.allure.kotlin.util.ResultsUtils` for replacement (as the implementation of parsing annotations has changed).
+ 
```

Comment:
Also, the constants which are listed below, is private now as it was initially in allure-java:
```
io.qameta.allure.android.io.TEXT_PLAIN
io.qameta.allure.android.io.TXT_EXTENSION
```